### PR TITLE
terraform-provider: validate image and microservice version

### PR DESCRIPTION
**Pull request by elchead**
### Context

This adds validation for the version support policy for the image and microservice versions to the provider.

### Proposed change(s)

-

### Checklist

- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [ ] Add labels (e.g., for changelog category)
- [ ] Is PR title adequate for changelog?
- [ ] Link to Milestone

### CODE REVIEW

File: terraform-provider-constellation/internal/provider/cluster_resource.go [hunk 1]
```diff
@@ -904,6 +895,26 @@ func (r *ClusterResource) apply(ctx context.Context, data *ClusterResourceModel,
 	return diags
 }
 
+func (r *ClusterResource) getImageVersion(ctx context.Context, data *ClusterResourceModel) (imageAttribute, semver.Semver, diag.Diagnostics) {
+	var image imageAttribute
+	diags := data.Image.As(ctx, &image, basetypes.ObjectAsOptions{})
```
Comment (msanft) [inline terraform-provider-constellation/internal/provider/cluster_resource.go:900]: We'd probably want to return early here, as when this fails (conversion error), the 2 subsequent calls are going to fail as well

File: terraform-provider-constellation/internal/provider/cluster_resource.go [hunk 2]
```diff
@@ -904,6 +895,26 @@ func (r *ClusterResource) apply(ctx context.Context, data *ClusterResourceModel,
...
+		diags.AddAttributeError(
+			path.Root("image").AtName("version"),
+			"Invalid image version",
+			fmt.Sprintf("Parsing image version (%s): %s", image.Version, err))
+		return imageAttribute{}, semver.Semver{}, diags
+	}
+	if err := compatibility.BinaryWith(r.providerData.Version.String(), imageSemver.String()); err != nil {
+		diags.AddAttributeError(
+			path.Root("image").AtName("version"),
+			"Image version incompatible with provider version",
```
Comment (msanft) [inline terraform-provider-constellation/internal/provider/cluster_resource.go:912]:
```suggestion
			"Invalid image version",
```

nit: align this with the error message for an invalid microservice version

File: terraform-provider-constellation/internal/provider/cluster_resource_test.go [hunk 3]
```diff
@@ -7,14 +7,104 @@ SPDX-License-Identifier: AGPL-3.0-only
...
+				Version: tc.version,
+			}
+
+			input, diags := basetypes.NewObjectValueFrom(context.Background(), map[string]attr.Type{
+				"version":    basetypes.StringType{},
+				"reference":  basetypes.StringType{},
+				"short_path": basetypes.StringType{},
+			}, img)
+			require.Equal(t, 0, diags.ErrorsCount())
+			_, _, diags2 := sut.getImageVersion(context.TODO(), &ClusterResourceModel{
```
Comment (msanft) [inline terraform-provider-constellation/internal/provider/cluster_resource_test.go:100]: q: Is there a reason why we use `context.TODO()` here?

Comment (elchead) [reply terraform-provider-constellation/internal/provider/cluster_resource_test.go:100]: no 🙈

**Review by msanft — APPROVED** [review]

**Review by derpsteb — APPROVED** [review]

**Pull request merged**